docs/developer/architecture/core/logging-service.asciidoc

```diff
@@ -0,0 +1,547 @@
+[[logging-service]]
+== Logging service
+Allows a plugin to provide status and diagnostic information.
+
+NOTE: The Logging service is only available server side. 
+
+[source,typescript]
+----
+import type { PluginInitializerContext, CoreSetup, Plugin, Logger } from 'kibana/server';
+
+export class MyPlugin implements Plugin {
+  private readonly logger: Logger;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
+  }
+
+  public setup(core: CoreSetup) {
+    try {
+      this.logger.debug('doing something...');
+      // …
+    } catch (e) {
+      this.logger.error('failed doing something...');
+    }
+  }
+}
+----
+
+The way logging works in {kib} is inspired by `log4j 2` logging framework used by {ref-bare}/current/logging.html[Elasticsearch].
+The main idea is to have consistent logging behavior (configuration, log format etc.) across the entire Elastic Stack where possible.
+
+=== Loggers, Appenders and Layouts
+
+{kib} logging system has three main components: _loggers_, _appenders_ and _layouts_. These components allow us to log
+messages according to message type and level, to control how these messages are formatted and where the final logs
+will be displayed or stored.
+
+__Loggers__ define what logging settings should be applied to a particular logger.
+
+__<<logging-appenders,Appenders>>__ define where log messages are displayed (eg. stdout or console) and stored (eg. file on the disk).
+
+__<<logging-layouts,Layouts>>__ define how log messages are formatted and what type of information they include.
+
+[[logger-hierarchy]]
+=== Logger hierarchy
```

Comment:
I wonder if layouts and appenders should be higher up in the doc here since I suspect that is what most users will want to customize. Configuring specific loggers is more of an advanced usage.